flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/LeaderElectionEventHandler.java

```diff
@@ -56,4 +74,18 @@ public interface LeaderElectionEventHandler {
      *     address.
      */
     void onLeaderInformationChange(LeaderInformation leaderInformation);
+
+    /**
+     * Called by specific {@link LeaderElectionDriver} when the leader information is changed. Then
+     * the {@link LeaderElectionService} could write the leader information again if necessary. This
+     * method should only be called when {@link LeaderElectionDriver#hasLeadership()} is true.
+     * Duplicated leader change events could happen, so the implementation should check whether the
+     * passed leader information is really different with internal confirmed leader information.
+     *
+     * <p>This method * will trigger the grant event processing in a separate thread.
```

Comment:
```suggestion
     * <p>This method will trigger the leader information change event processing in a separate thread.
```

Reply:
Oh man :facepalm: Thanks for being precise in your review :+1: